server/schemas/schemaController.js

```diff
@@ -36,6 +38,7 @@ var handleUpload = function(request, response, appendData) {
       validateEntry(schemaName, buffer, function(error, isValid) {
```

Comment:
why not just call directly into `schemaMatcher`? `validateEntry` seems like unnecessary redirection.